org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceDiagnosticsHandler.java

```diff
@@ -119,6 +121,9 @@ public void resourceChanged(IResourceChangeEvent event) {
 	 */
 	@Override
 	public boolean visit(IResourceDelta delta) throws CoreException {
+		if (delta.getFlags() == IResourceDelta.MARKERS && Arrays.stream(delta.getMarkerDeltas()).map(IMarkerDelta::getMarker).noneMatch(WorkspaceDiagnosticsHandler::isInteresting)) {
```

Comment:
Should the check be ((delta.getFlags() & IResourceDelta.MARKERS) == 1) in case there are additional flags embedded ? (it's a bitmask).

Reply:
I don't think so; the goal here is to skip the visit only if it's about an ignored marker change. If the change is combined with another type of change (eg content), it's probably still worth visiting it.